### PR TITLE
chore(site): move ssh keys management to react-query

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

File: site/src/api/queries/sshKeys.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export const useUserSSHKey = (userId: string) => {
+  return useQuery({
+    queryKey: getUserSSHKeyQueryKey(userId),
+    queryFn: () => API.getUserSSHKey(userId),
+  });
+};
+
+export const useRegenerateUserSSHKey = (
+  userId: string,
+  onSuccess: () => void,
```
Comment (aslilac) [inline site/src/api/queries/sshKeys.ts:15]: see!!! it was already going to leak in!! 😝

File: site/src/pages/UserSettingsPage/SSHKeysPage/SSHKeysPage.tsx [hunk 2]
```diff
@@ -15,40 +15,40 @@ export const Language = {
...
-        open={sshState.matches("confirmSSHKeyRegenerate")}
-        confirmLoading={sshState.matches("regeneratingSSHKey")}
+        open={isConfirmingRegeneration}
+        confirmLoading={regenerateUserSSHKeyMutation.isLoading}
         title={Language.regenerateDialogTitle}
         confirmText={Language.confirmLabel}
         onConfirm={() => {
-          sshSend({ type: "CONFIRM_REGENERATE_SSH_KEY" });
+          regenerateUserSSHKeyMutation.mutate();
         }}
```
Comment (aslilac) [inline site/src/pages/UserSettingsPage/SSHKeysPage/SSHKeysPage.tsx:49]:
```suggestion
onConfirm={regenerateUserSSHKeyMutation.mutate}
```

would this work? or does `mutate` need `this`?

File: site/src/xServices/sshKey/sshKeyXService.ts
Comment (aslilac) [inline site/src/xServices/sshKey/sshKeyXService.ts:1]: bye bye!!

**Review by aslilac — APPROVED** [review]
one small suggestion, otherwise gtg!

File: site/src/pages/UserSettingsPage/SSHKeysPage/SSHKeysPage.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/pages/UserSettingsPage/SSHKeysPage/SSHKeysPage.tsx:49]: Yeap, it works!

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/9598

**Pull request merged**